### PR TITLE
Remove d3 dependency in crossfilter

### PR DESCRIPTION
**Pull request by mweiden**
Fixes https://github.com/chanzuckerberg/cellxgene/issues/648

### CODE REVIEW

File: client/src/util/typedCrossfilter/crossfilter.js [hunk 1]
```diff
@@ -11,6 +9,8 @@ import {
 } from "./sort";
 import { makeSortIndex } from "./util";
 
+const pointInPolygon = require("point-in-polygon");
```
Comment (colinmegill) [inline client/src/util/typedCrossfilter/crossfilter.js]: `require` --> `import`

**Comment by colinmegill** [comment]
Hi @mweiden! 

Could you tell me a little more about this PR? I would expect we would move off of the d3 dependency to another package if we were trying to get off of d3 altogether in the app (I don't think that's a requirement as of now), or if this package is significantly faster.

Otherwise, I would prefer we not merge this, and remain on the d3 version.

**Comment by colinmegill** [comment]
(Otherwise,  we're just adding a different dependency, which — I'm unsure if that was @bkmartinjr's intention)

**Comment by mweiden** [comment]
@colinmegill pardon, I misunderstood the ask in the issue. @bkmartinjr and I just sync'ed and came up with the following plan:

1. Lift [the code referenced in the issue](https://github.com/substack/point-in-polygon/blob/master/index.js) directly into the module (with attribution); add tests.
1. Merge this PR (approval and coarse performance check pending).
1. At a later date, optimize the performance of the code, if necessary.

Thoughts?

**Comment by colinmegill** [comment]
That's what I thought he meant, sounds great. So long as there's no regression in performance from d3, sounds good to me.

**Pull request merged**